Microsoft.Toolkit.Uwp.UI.Animations/Behaviors/FadeHeaderBehavior.cs

```diff
@@ -0,0 +1,173 @@
+        {
+            var b = d as FadeHeaderBehavior;
+            b?.AssignFadeAnimation();
+        }
+
+        /// <summary>
+        /// The UIElement that will be faded.
+        /// </summary>
+        public static readonly DependencyProperty HeaderElementProperty = DependencyProperty.Register(
+            nameof(HeaderElement), typeof(UIElement), typeof(FadeHeaderBehavior), new PropertyMetadata(null, PropertyChangedCallback));
+
+        /// <summary>
+        /// Gets or sets the target element for the Fading behavior.
+        /// </summary>
+        /// <remarks>
+        /// Set this using the header of a ListView or GridView. You can use the entire root of the header or an element within the header.
+        ///
+        /// Using this example Header:
+        /// <ListView.Header>
+        ///     <Grid Name="MyHeader">
+        ///     </Grid>
+        /// </ListView.Header>
+        ///
+        /// The behavior would be implemented like this
+        /// <FadeHeaderBehavior HeaderElement="{Binding ElementName=HeaderPanel}" />
+        /// </remarks>
+        public UIElement HeaderElement
+        {
+            get { return (UIElement)GetValue(HeaderElementProperty); }
+            set { SetValue(HeaderElementProperty, value); }
+        }
+
+        /// <summary>
+        /// Called when the associated object is loaded.
+        /// </summary>
+        /// <param name="sender">The associated object</param>
+        /// <param name="routedEventArgs">The <see cref="RoutedEventArgs"/> instance containing the event data</param>
+        private void AssociatedObjectOnLoaded(object sender, RoutedEventArgs routedEventArgs)
+        {
+            AssignFadeAnimation();
+        }
+
+        /// <summary>
+        /// Uses Composition API to get the UIElement and sets an ExpressionAnimation
+        /// The ExpressionAnimation uses the height of the UIElement to calculate an opacity value
+        /// for the Header as it is scrolling off-screen. The opacity reaches 0 when the Header
+        /// is entirely scrolled off.
+        /// </summary>
+        private void AssignFadeAnimation()
+        {
+            // Confirm that Windows.UI.Xaml.Hosting.ElementCompositionPreview is available (Windows 10 10586 or later).
+            if (!ApiInformation.IsMethodPresent("Windows.UI.Xaml.Hosting.ElementCompositionPreview", nameof(ElementCompositionPreview.GetScrollViewerManipulationPropertySet)))
+            {
+                return;
+            }
+
+            if (AssociatedObject == null)
+            {
+                return;
+            }
+
+            var scroller = AssociatedObject as ScrollViewer ?? AssociatedObject.FindDescendant<ScrollViewer>();
+
+            if (scroller == null)
+            {
+                return;
+            }
+
+            // Implicit detection of a header if AssociatedObject is, or derived from, a ListView
+            if (HeaderElement == null)
```

Comment:
These two if statements can be merged and see whether AssociatedObject is ListViewBase which both ListView and GridView are. It's ListViewBase which has the Header property on it.